src/test/java/net/consensys/orion/http/handler/RetrievePrivacyGroupHandlerTest.java

```diff
@@ -0,0 +1,168 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+import net.consensys.cava.crypto.sodium.Box;
+import net.consensys.orion.enclave.Enclave;
+import net.consensys.orion.enclave.PrivacyGroupPayload;
+import net.consensys.orion.enclave.sodium.MemoryKeyStore;
+import net.consensys.orion.enclave.sodium.SodiumEnclave;
+import net.consensys.orion.helpers.FakePeer;
+import net.consensys.orion.http.handler.privacy.DeletePrivacyGroupRequest;
+import net.consensys.orion.http.handler.privacy.PrivacyGroup;
+import net.consensys.orion.http.handler.privacy.PrivacyGroupRequest;
+import net.consensys.orion.http.handler.privacy.RetrievePrivacyGroupRequest;
+import net.consensys.orion.utils.Serializer;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.security.SecureRandom;
+import java.util.Collections;
+
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.RecordedRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RetrievePrivacyGroupHandlerTest extends HandlerTest {
+
+  private static final String PRIVACY_GROUP_NAME = "test";
+  private static final String PRIVACY_GROUP_DESCRIPTION = "desc";
+
+  private final MemoryKeyStore memoryKeyStore = new MemoryKeyStore();
+  private String privacyGroupId;
+  private FakePeer peer;
+  private Box.PublicKey senderKey;
+  private String[] privacyGroupMembers;
+
+  @Override
+  protected Enclave buildEnclave(final Path tempDir) {
+    memoryKeyStore.generateKeyPair();
+    return new SodiumEnclave(memoryKeyStore);
+  }
+
+  @BeforeEach
+  void setup() throws IOException, InterruptedException {
+    senderKey = memoryKeyStore.generateKeyPair();
+    final Box.PublicKey recipientKey = memoryKeyStore.generateKeyPair();
+
+    privacyGroupMembers = new String[] {encodeBytes(senderKey.bytesArray()), encodeBytes(recipientKey.bytesArray())};
+    final PrivacyGroupRequest privacyGroupRequestExpected = buildPrivacyGroupRequest(
+        privacyGroupMembers,
+        encodeBytes(senderKey.bytesArray()),
+        PRIVACY_GROUP_NAME,
+        PRIVACY_GROUP_DESCRIPTION);
+    final Request request = buildPrivateAPIRequest(CREATE_PRIVACY_GROUP, JSON, privacyGroupRequestExpected);
+
+    final byte[] privacyGroupId = enclave.generatePrivacyGroupId(
+        new Box.PublicKey[] {senderKey, recipientKey},
+        privacyGroupRequestExpected.getSeed().get(),
+        PrivacyGroupPayload.Type.PANTHEON);
+
+    // create fake peer
+    peer = new FakePeer(new MockResponse().setBody(encodeBytes(privacyGroupId)), recipientKey);
+    networkNodes.addNode(Collections.singletonList(peer.publicKey), peer.getURL());
+
+    // execute request
+    final Response resp = httpClient.newCall(request).execute();
+
+    final RecordedRequest recordedRequest = peer.server.takeRequest();
+    assertThat(recordedRequest.getPath()).isEqualTo(PUSH_PRIVACY_GROUP);
```

Comment:
nit: asserts during setup seems odd....
[confirming this whole setup is to create a PrivacyGroup in a node?] - If so - can we break it all out to a function called "createPrivacyGroup"? - Then invoke that from the "beforeEach" (or otherwise just annotate the function).

Reply:
This is all to create a privacy group. Split into own method. Didn't add annotate with beforeEach as this needs to happen after the keys created in the setup and the order of beforeEach methods are not deterministic.